backend/errors/index.ts

```diff
@@ -0,0 +1,8 @@
+export class ApiError extends Error {
```

Comment:
I just added this because for proper rest api you need to have status codes and when you throw errors, you only do it with 500 response but the reasons of the errors often determine what status code you need so what you want instead is to throw error with appropriate status code and then send that status code as a response